v1/src/test/java/com/google/cloud/teleport/templates/PubSubToPubSubIT.java

```diff
@@ -0,0 +1,116 @@
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Integration test for {@link PubsubToPubsub} (Cloud_PubSub_to_Cloud_PubSub). */
+@Category(TemplateIntegrationTest.class)
+@TemplateIntegrationTest(PubsubToPubsub.class)
+@RunWith(JUnit4.class)
+public class PubSubToPubSubIT extends TemplateTestBase {
+  private static PubsubResourceManager pubsubResourceManager;
+
+  @Before
+  public void setUp() throws IOException {
+    pubsubResourceManager =
+        DefaultPubsubResourceManager.builder(testName.getMethodName(), PROJECT)
+            .credentialsProvider(credentialsProvider)
+            .build();
+  }
+
+  @After
+  public void tearDown() {
+    pubsubResourceManager.cleanupAll();
+  }
+
+  @Test
+  public void testSubscriptionToTopic() throws IOException {
+    // Arrange
+    String jobName = createJobName(testName.getMethodName());
+    TopicName inputTopic = pubsubResourceManager.createTopic("input-topic");
+    TopicName outputTopic = pubsubResourceManager.createTopic("output-topic");
+    SubscriptionName inputSubscription =
+        pubsubResourceManager.createSubscription(inputTopic, "input-subscription");
+    SubscriptionName outputSubscription =
+        pubsubResourceManager.createSubscription(outputTopic, "output-subscription");
+
+    List<String> messages =
+        List.of("message1-" + jobName, "message2-" + jobName, "message3-" + jobName);
+    publishMessages(inputTopic, messages);
+    LaunchConfig.Builder options =
+        LaunchConfig.builder(jobName, specPath)
+            .addParameter("inputSubscription", inputSubscription.toString())
+            .addParameter("outputTopic", outputTopic.toString());
+
+    // Act
+    JobInfo info = launchTemplate(options);
+    assertThat(info.state()).isIn(JobState.ACTIVE_STATES);
+
+    AtomicReference<PullResponse> records = new AtomicReference<>();
+    Result result =
+        new DataflowOperator(getDataflowClient())
+            .waitForConditionAndFinish(
+                createConfig(info),
+                () -> {
+                  PullResponse response =
+                      pubsubResourceManager.pull(outputSubscription, messages.size());
+                  records.set(response);
+                  return response.getReceivedMessagesList().size() == messages.size();
+                });
+
+    // Assert
+    List<String> actualMessages =
+        records.get().getReceivedMessagesList().stream()
+            .map(receivedMessage -> receivedMessage.getMessage().getData().toStringUtf8())
+            .collect(Collectors.toList());
+    assertThat(result).isEqualTo(Result.CONDITION_MET);
+    assertThat(actualMessages).containsExactlyElementsIn(messages);
```

Comment:
We may want to use our own assertion that deduplicates, but that can be hardened later IMO

Reply:
Changed the assertion to only check if all expected messages are present in the actual messages